File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/models/kafkadlq/SedSendtHendelseKafkaDLQ.java
package no.nav.melosys.eessi.models.kafkadlq;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.Data;
import no.nav.melosys.eessi.kafka.consumers.SedHendelse;
import org.hibernate.annotations.Type;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;

@Entity
@Data
@DiscriminatorValue("SED_SENDT_HENDELSE")
public class SedSendtHendelseKafkaDLQ extends KafkaDLQ {

    @Type(type = "jsonb")
    @Column(name = "melding", columnDefinition = "jsonb")
    private SedHendelse sedSendtHendelse;

    @Override
    public String hentMeldingSomStreng() throws JsonProcessingException {
        ObjectMapper objectMapper = new ObjectMapper();
        return objectMapper.writeValueAsString(sedSendtHendelse);
    }
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/models/kafkadlq/SedSendtHendelseKafkaDLQ.kt
package no.nav.melosys.eessi.models.kafkadlq

import com.fasterxml.jackson.databind.ObjectMapper
import jakarta.persistence.Column
import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity
import no.nav.melosys.eessi.kafka.consumers.SedHendelse
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes

@Entity
@DiscriminatorValue("SED_SENDT_HENDELSE")
class SedSendtHendelseKafkaDLQ : KafkaDLQ() {
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "melding", columnDefinition = "jsonb")
    var sedSendtHendelse: SedHendelse? = null

    override fun hentMeldingSomStreng(): String = ObjectMapper().writeValueAsString(sedSendtHendelse)
}
